content/doc/book/managing/plugins.adoc

```diff
@@ -4,13 +4,257 @@ layout: section
+plugin updates.
+
+=== Uninstalling a plugin
+
+The simplest way to uninstall a plugin is to navigate to the *Installed* tab on
+the *Manage Plugins* page. From there, Jenkins will automatically determine
+which plugins are safe to uninstall, those which are not dependencies of other
+plugins, and present a button for doing so.
+
+image::managing/plugin-manager-uninstall.png["Installed tab in the Plugin Manager", role=center]
+
+
+A plugin may also be uninstalled by removing the corresponding `.hpi`
+file from the `JENKINS_HOME/plugins` directory on the master. The plugin will
+continue to function until the master has been restarted.
+
+[CAUTION]
+====
+If a plugin `.hpi` file is removed but required by other plugins, the Jenkins
+master may fail to boot correctly.
+====
+
+Uninstalling a plugin does *not* remove the configuration that the plugin may
+have created. If there are existing jobs/nodes/views/builds/etc configurations
+that reference data created by the plugin, during boot Jenkins will warn that
+some configurations could not be fully loaded and ignore the unrecognized data.
+
+Since the configuration(s) will be preserved until they are overwritten,
+re-installing the plugin will result in those configuration values reappearing..
+
+==== Removing old data
+
+Jenkins provides a facility for purging configuration left behind by
+uninstalled plugins. Navigate to *Manage Jenkins* and then click on *Manage
+Old Data* to review and remove old data.
+
+=== Disabling a plugin
+
+Disabling a plugin is a softer way to retire a plugin. Jenkins will continue to
+recognize that the plugin is installed, but it will not start the plugin, and
+no extensions contributed from this plugin will be visible.
+
+A Jenkins administrator may disable a plugin by unchecking the box on the
+*Installed* tab of the *Manage Plugins* page (see below).
+
+image::managing/plugin-manager-disable.png["Installed tab in the Plugin Manager", role=center]
+
+
+A systems administrator may also disable a plugin by creating a file on the
+Jenkins master, such as: `JENKINS_HOME/plugins/PLUGIN_NAME.hpi.disabled`.
+
+The configuration(s) created by the disabled plugin behave as if the plugin
+were uninstalled, insofar that they result in warnings on boot but are
+otherwise ignored.
+
+
+== Pinned plugins
+
+[CAUTION]
+====
+Pinned plugins feature was removed in Jenkins 2.0. Versions later than Jenkins
+2.0 do not bundle plugins, instead providing a wizard to install the most
+useful plugins.
 ====
+
+The notion of *pinned plugins* applies to plugins that are bundled with
+Jenkins 1.x, such as the
+link:https://plugins.jenkins.io/matrix-auth[*Matrix Authorization plugin*].
+
+By default, whenever Jenkins is upgraded, its bundled plugins overwrite the
+versions of the plugins are currently installed in `JENKINS_HOME`.
+
+However, when a bundled plugin has been manually updated, Jenkins will mark
+that plugin as pinned to the particular version. On the file system, Jenkins
```

Comment:
"filesystem" isn't a word AFAICT, everywhere else I have been using "file system" (such as in the glossary